src/Components/PostComment.jsx

```diff
@@ -0,0 +1,55 @@
+import { useContext, useState } from "react"
+import { postComment } from "../api"
+import { UserContext } from "../Context/UserContext";
+
+const PostComment = ({ article_id , setCommentsList}) => {
+    const { user } = useContext(UserContext);
+  const [isLoading, setIsLoading] = useState(false)
+  const [isErr, setIsErr] = useState(false)
+
+    const [newComment, setNewComment] = useState('')
+    const [blankComment, setBlankComment] = useState({id: '', placeholder: 'Comment...'})
+
+    const handleInput = (event) => {
+        setNewComment(event.target.value)
+    }
+    const handleSubmit = (event) => {
+        event.preventDefault();
+      if (newComment) {
+          setIsLoading(true)
+          postComment(article_id, { username: user.username, body: newComment }).then((postedComment) => {
+            setCommentsList((currComments) => {
+                  return [postedComment, ...currComments]
+              })
+              setNewComment('')
+            setBlankComment({ id: '', placeholder: 'Comment...' }) 
+            setIsLoading(false)
+          }).catch((err) => {
+              setIsErr(true)
+            })
+
```

Comment:
I like that you've got some input validation in place
Currently though the user can mash the button and enter the same comment multiple times while it is posting (something they may be likely to do if their connection is slow)
You could make it so the user is unable to post while a comment is currently in the process of being posted (by disabling the post button for example)